base/sparse/sparsevector.jl

```diff
@@ -1629,3 +1630,25 @@ function sort{Tv,Ti}(x::SparseVector{Tv,Ti}; kws...)
     newnzvals = allvals[deleteat!(sinds[1:k],z)]
     SparseVector(n,newnzind,newnzvals)
 end
+
+#set-like methods
+function union!{Tv,Ti<:Integer}(s::Set,x::SparseVector{Tv,Ti})
+    vals = nonzeros(x)
+    s = union!(s,vals)
+    if length(x) == length(vals)
+        push!(s,zero(Tv))
+    end
+    s
+end
+function unique{Tv,Ti<:Integer}(x::SparseVector{Tv,Ti})
+    inds = sort(x.nzind)
```

Comment:
generally these should always be sorted as part of the data structure convention